jena-rdfconnection/src/main/java/org/apache/jena/rdflink/RDFLinkHTTP.java

```diff
@@ -241,77 +243,111 @@ public QueryExec query(Query query) {
 
     @Override
     public QueryExecBuilder newQuery() {
-        return createQExecBuilder();
+        return createQExecBuilder(null);
     }
 
     // Create the QExec
 
     private QueryExec queryExec(Query query, String queryString, QueryType queryType) {
-        checkQuery();
-        if ( query == null && queryString == null )
-            throw new InternalErrorException("Both query and query string are null");
-        if ( query == null ) {
-            if ( parseCheckQueries )
-                // Don't retain the query.
-                QueryFactory.create(queryString);
```

Comment:
Something has changed because this:
```
    public static void main(String... args) {
        RDFLink link = RDFLinkHTTP.service("http://localhost:3030/ds").parseCheckSPARQL(true).build();
        try ( link; ) {
            link.query("Custom");
        }
        System.out.println("Done");
    }
```
throws an exception in Jena5/main but not in this PR. The parse error has moved to the point of execution.

Reply:
Done.